src/hooks/useSignInForm.ts

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { signin } from '../apis/auth';
+import { ERROR_MESSAGES } from '../constants/errorMessages';
+import { ROUTES } from '../constants/routes';
+import { validateSignInInput } from './../utils/validations';
+
+interface InitialState {
+  email: string;
+  password: string;
+}
+
+const useSignInForm = (initialState: InitialState) => {
+  const [values, setValues] = useState(initialState);
+  const [errors, setErrors] = useState(initialState);
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { name, value } = event.target;
+    setValues({ ...values, [name]: value });
+    setErrors({ ...errors, [name]: '' });
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
사소한데 useState에서 react import하고 있으니 FormEvent로 같이 포함해도 될 것 같아용 가볍게 참고만 해주세요!
```suggestion
  import { FormEvent, useState } from 'react';
  // ...
  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
```

Reply:
오 좋아요, 이것도 코드 짜면서 어떤게 더 좋을까 생각했었는데 말씀해주신 것처럼 하는 게 더 간결해지겠네요. 기억해뒀다가 나중에 리팩터링할 때 얘기 꺼내겠습니다!